File: platforms/android/foundation/foundation/src/main/java/io/nativeblocks/foundation/modifier/Padding.kt
package io.nativeblocks.foundation.modifier

import androidx.compose.foundation.layout.padding
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import io.nativeblocks.compiler.type.Modifier
import io.nativeblocks.compiler.type.ModifierData
import androidx.compose.ui.Modifier as UiModifier

@Modifier(
    keyType = "nativeblocks/padding",
    name = "Padding",
    description = "Adds padding around the block.",
    version = 1,
    versionName = "1",
)
internal fun padding(
    @ModifierData(description = "Padding on the start side in DP.", defaultValue = "0.0")
    start: Dp = 0.dp,
    @ModifierData(description = "Padding on the top side in DP.", defaultValue = "0.0")
    top: Dp = 0.dp,
    @ModifierData(description = "Padding on the end side in DP.", defaultValue = "0.0")
    end: Dp = 0.dp,
    @ModifierData(description = "Padding on the bottom side in DP.", defaultValue = "0.0")
    bottom: Dp = 0.dp,
): UiModifier = UiModifier.padding(start = start, top = top, end = end, bottom = bottom)

File: platforms/ios/foundation/Sources/NativeblocksFoundation/modifier/Padding.swift
import NativeblocksCompiler
import NativeblocksRuntime
import SwiftUI

@Modifier(
    name: "Padding",
    keyType: "nativeblocks/padding",
    description: "Adds padding around the block.",
    version: 1,
    versionName: "1"
)
struct Padding: ViewModifier {

    @ModifierData(description: "Padding on the leading side.", defaultValue: "0.0")
    var leading: CGFloat = 0

    @ModifierData(description: "Padding on the top side.", defaultValue: "0.0")
    var top: CGFloat = 0

    @ModifierData(description: "Padding on the trailing side.", defaultValue: "0.0")
    var trailing: CGFloat = 0

    @ModifierData(description: "Padding on the bottom side.", defaultValue: "0.0")
    var bottom: CGFloat = 0

    func body(content: Content) -> some View {
        content
            .padding(.leading, leading)
            .padding(.top, top)
            .padding(.trailing, trailing)
            .padding(.bottom, bottom)
    }
}
